core/src/main/java/io/parsingdata/metal/expression/comparison/ComparisonExpression.java

```diff
@@ -57,7 +59,7 @@ public ComparisonExpression(final ValueExpression value, final ValueExpression p
 
     @Override
     public boolean eval(final ParseGraph graph, final Encoding encoding) {
-        final ImmutableList<Optional<Value>> values = value == null ? ImmutableList.create(Optional.of(graph.current())) : value.eval(graph, encoding);
+        final ImmutableList<Optional<Value>> values = value == null ? ImmutableList.create(graph.current().map(identity())) : value.eval(graph, encoding);
```

Comment:
I was thinking along similar lines when editing this code. I don't think there is a clean default such ad `self` here because sometimes the `null` really means _nothing_.

But your suggestion is quite interesting: we could perhaps create an _empty_ expression that throws an exception if you `eval()` it, but exists solely to signal that it formally doesn't exist. Still, the end result would be similar to a `null` (exception thrown when not checking). Worth exploring!